### PR TITLE
Fix GFX_PAYLOAD handling (bsc#973144)

### PR DESCRIPTION
**Pull request by k0da**
Signed-off-by: Dinar Valeev dvaleev@suse.com

### CODE REVIEW

**Comment by k0da** [comment]
No, still wrong...

# Uncomment to disable graphical terminal (grub-pc only)
#GRUB_TERMINAL=console
# The resolution used on graphical terminal
# note that you can use only modes which your graphic card supports via VBE
# you can see them in real GRUB with the command `vbeinfo'
#GRUB_GFXMODE=auto
# Uncomment if you don't want GRUB to pass "root=UUID=xxx" parameter to Linux
#GRUB_DISABLE_LINUX_UUID=true
# Uncomment to disable generation of recovery mode menu entries
GRUB_DISABLE_RECOVERY=true
# Uncomment to get a beep at grub start
#GRUB_INIT_TUNE="480 440 1"
# Skip 30_os-prober if you experienced very slow in probing them
# WARNING foregin OS menu entries will be lost if set true here
GRUB_DISABLE_OS_PROBER=false
# Set to 'y' for grub to be installed on an encrypted partition
GRUB_ENABLE_CRYPTODISK=n
# Set to 'text' to avoid loading graphical modules
GFX_PAYLOAD_LINUX=keep
SUSE_BTRFS_SNAPSHOT_BOOTING=true

File: src/Core/GRUB2.pm [hunk 1]
```diff
@@ -824,7 +824,7 @@ sub Global2Info {
         } elsif ($key =~ m/@?GRUB_ENABLE_CRYPTODISK$/) {
             $ret{"cryptodisk"} = $val eq 'y' ? 1 : 0;
         } elsif ($key =~ m/@?GRUB_GFXPAYLOAD_LINUX$/) {
-            $ret{"gfxterm"} = ($val eq "console") ? "text" : "keep"
+            $ret{"payload"} = ($val eq "console") ? "text" : "keep"
```
Comment (wfeldt) [inline src/Core/GRUB2.pm]: What happens with `$ret{payload}`?

**Comment by wfeldt** [comment]
Could you please clarify what values GRUB_GFXPAYLOAD_LINUX and $terminal are expected to have? the yast code seems to expect $terminal to be either console or gfxterm.

**Comment by k0da** [comment]
If terminal is set to serial/console then GRUB_GFXPAYLOAD_LINUX should be text
else GRUB_GFXPAYLOAD_LINUX is unset or set to keep

**Comment by wfeldt** [comment]
the code path is not used anymore, closing

**Pull request closed without merging**